File: roster-builder-front/src/app/Entities/Role.ts
import { Unit } from './Unit';
import { SpecialAbilities } from './SpecialAbilities';

export class Role {
  public faction?: string;
  public name?: string;
  public description?: string;
  public limit?: number;
  public specialAbilities?: SpecialAbilities[];
  public unit?: Unit;
}

File: WarhammerAPI/WarhammerAPI/Model/DAL/Entities/Role.cs
﻿using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Threading.Tasks;

namespace WarhammerAPI.Model.DAL.Entities
{
    public class Role
    {
        [Key]
        public int Id { get; set; }
        public string Faction { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public int Limit { get; set; }
        public List<SpecialAbilities> SpecialAbilities { get; set; }
        public Unit Unit { get; set; }
    }
}
